modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/search/filter/FilterUtil.java

```diff
@@ -14,40 +14,245 @@
+			_addTransformedFilters(
+				booleanFilter.getMustNotBooleanClauses(),
+				BooleanClauseOccur.MUST_NOT, transformedBooleanFilter);
+			_addTransformedFilters(
+				booleanFilter.getShouldBooleanClauses(),
+				BooleanClauseOccur.SHOULD, transformedBooleanFilter);
 
-		if (!termFilterField.startsWith(DDMIndexer.DDM_FIELD_PREFIX)) {
-			return filter;
+			return transformedBooleanFilter;
+		}
+
+		@Override
+		public Filter visit(DateRangeTermFilter dateRangeTermFilter) {
+			return dateRangeTermFilter;
+		}
+
+		@Override
+		public Filter visit(ExistsFilter existsFilter) {
+			return existsFilter;
+		}
+
+		@Override
+		public Filter visit(GeoBoundingBoxFilter geoBoundingBoxFilter) {
+			return geoBoundingBoxFilter;
+		}
+
+		@Override
+		public Filter visit(GeoDistanceFilter geoDistanceFilter) {
+			return geoDistanceFilter;
+		}
+
+		@Override
+		public Filter visit(GeoDistanceRangeFilter geoDistanceRangeFilter) {
+			return geoDistanceRangeFilter;
+		}
+
+		@Override
+		public Filter visit(GeoPolygonFilter geoPolygonFilter) {
+			return geoPolygonFilter;
+		}
+
+		@Override
+		public Filter visit(MissingFilter missingFilter) {
+			return missingFilter;
+		}
+
+		@Override
+		public Filter visit(PrefixFilter prefixFilter) {
+			return prefixFilter;
+		}
+
+		@Override
+		public Filter visit(QueryFilter queryFilter) {
+			Query query = queryFilter.getQuery();
+
+			if (query instanceof WildcardQuery) {
+				WildcardQuery wildcardQuery = (WildcardQuery)query;
+
+				QueryTerm queryTerm = wildcardQuery.getQueryTerm();
+
+				String queryTermFieldName = queryTerm.getField();
+
+				if (!queryTermFieldName.startsWith(
+						DDMIndexer.DDM_FIELD_PREFIX)) {
+
+					return queryFilter;
+				}
+
+				DDMStructureField ddmStructureField = DDMStructureField.from(
+					queryTermFieldName);
+
+				try {
+					return _createDDMStructureQueryFilter(
+						ddmStructureField.getDDMStructureFieldName(),
+						new WildcardQueryImpl(
+							ddmStructureField.
+								getDDMStructureNestedTypeSortableFieldName(),
+							queryTerm.getValue()));
+				}
+				catch (Exception e) {
+					return queryFilter;
+				}
+			}
+
+			return queryFilter;
+		}
+
+		@Override
+		public Filter visit(RangeTermFilter rangeTermFilter) {
+			String rangeTermFilterFieldName = rangeTermFilter.getField();
+
+			if (!rangeTermFilterFieldName.startsWith(
+					DDMIndexer.DDM_FIELD_PREFIX)) {
+
+				return rangeTermFilter;
+			}
+
+			DDMStructureField ddmStructureField = DDMStructureField.from(
+				rangeTermFilterFieldName);
+
+			try {
+				return _createDDMStructureQueryFilter(
+					ddmStructureField.getDDMStructureFieldName(),
+					new TermRangeQueryImpl(
+						ddmStructureField.
+							getDDMStructureNestedTypeSortableFieldName(),
+						rangeTermFilter.getLowerBound(),
+						rangeTermFilter.getUpperBound(),
+						rangeTermFilter.isIncludesLower(),
+						rangeTermFilter.isIncludesUpper()));
+			}
+			catch (Exception exception) {
+				return rangeTermFilter;
```

Comment:
this is strange

Reply:
I've considered that it is better to return the original filter in case there are some problem creating the new one but I will give it a second thought